docs/explanation.rst

```diff
@@ -0,0 +1,121 @@
+Explanantion
+=============
+
+..
+  This section requires more improvements.
+
+
+Upload Sets
+-----------
+``UploadSet`` is The main object provided by this extension. Each `UploadSet` is
+responsible for dealing with set of files. You declare `UploadSet` in the code::
+
+    photos = UploadSet('photos', IMAGES)
+
+And then you can use the `UploadSet.save` method to save uploaded files and
+`UploadSet.path` and `UploadSet.url` to access them.
+
+during saving & retrieving the uploaded file, `UploadSet` perform some security
+checks for you.
+
+
+Security Checks:
+----------------
+
+Security itself is a huge subject and you should consider designing it
+seriously. This extension is not intended by first intention to protect your
+app. However ``Flask-Reuploaded`` extension tries to help you in preveting the
+common simple attacks or user errors. for Example::
+
+    - User can't upload file with filename that starts with `.` 
+      This means that user can't upload `.`, '..', '../../../home/{username}/.bash' or '../../my_app.wsgi'.
+      i.e:. filename will cahnged or internal server error will be raised.
```

Comment:
```suggestion
      e.g. `.`, '..', '../../../home/{username}/.bash' or '../../my_app.wsgi'.
      When a user tries to upload a file with such a filename, the file will be renamed.
```

What is the thing with the internal server error?

In which cases will it be raised? If you know one, we need to be more specific.